### ISSUE TITLE
Weird-looking distribution of `Muon::isolationR03().emEt` vs muon  ϕ- in Phase 2 samples

### ISSUE THREAD

**Issue by mmusich** [cms-sw/cmssw#43858]
While studying the Phase2 TkAl ALCARECO events produced in the `Phase2Fall22DRMiniAOD` campaign, a peculiar feature was noticed in the ϕ distribution of the tracks from Z → µµ decays selected from alignment (see [here](https://indico.cern.ch/event/1363142/contributions/5787810/attachments/2789909/4865144/cmsal_240130.pdf#page=12) for more details).
Upon dedicated check at the level of ALCARECO sample, it was observed that the feature was already present in the tracks persisted in the input sample, but also in the muon tracks from W → μν decays. 
Checking on the input muons, the reason of the ϕ-dependent inefficiency was spotted in an abnormal distribution of the muon electromagnetic relative isolation distribution `Muon::isolationR03().emEt  as a function of the muon azimuth. This quantity is used together with the tracker and hadronic relative isolation to select the muons entering the alignment samples.

https://github.com/cms-sw/cmssw/blob/f2ffd7b3c9b926a2db457de7749d4fca2183ba53/Alignment/CommonAlignmentProducer/python/TkAlMuonSelectors_cfi.py#L16

For reference it is how the `isolationR03().emEt` looks like in a **NO PU** SingleMuon gun [*] (hence I would naively expect it to be close to zero):

![Screenshot from 2024-02-03 18-56-30](https://github.com/cms-sw/cmssw/assets/5082376/c6d6fc0a-407c-44d8-be42-b78fd0f283c5)

And here the full relative combined isolation vs muon 𝜂-ϕ.  

![Screenshot from 2024-02-03 18-56-54](https://github.com/cms-sw/cmssw/assets/5082376/35709af9-8036-4d06-bdd2-310810a4e0ba)

[*]
```
 /RelValSingleMuFlatPt2To100/CMSSW_14_0_0_pre2-133X_mcRun4_realistic_v1_STD_2026D98_noPU_RV229-v1/GEN-SIM-RECO 
```

**Comment by mmusich** [comment]
type muon

**Comment by slava77** [comment]
wouldn't it be more practical to use PF isolation?
CaloTower-based isolation in phase-2 isn't supposed to work in the endcaps; and from this issue it seems like the barrel is not healthy either.

**Comment by mmusich** [comment]
> wouldn't it be more practical to use PF isolation?

I was told to use tracker isolation instead.

**Comment by slava77** [comment]
> > wouldn't it be more practical to use PF isolation?
> 
> I was told to use tracker isolation instead.

track iso should work OK

**Comment by mmusich** [comment]
> track iso should work OK

yes, but the point of this issue is that this minefield is left in release for anyone to step into. Something should be changed to avoid returning nonsense.

**Comment by makortel** [comment]
assign alca, reconstruction, upgrade

**Comment by jfernan2** [comment]
FYI
@24LopezR @rbhattacharya04

**Comment by rbhattacharya04** [comment]
Since the issue is with the electromagnetic component of the isolation, it would be nice if Egamma can check whether they see a similar trend in their isolation variable.

**Comment by mmusich** [comment]
@cms-sw/ecal-dpg-l2 FYI

**Comment by swagata87** [comment]
It looks like electrons and photons don't have this issue. For example, I attach one plot for electrons in barrel. 
Sample used: `/RelValZEE_14/CMSSW_14_0_0_pre3-140X_mcRun4_realistic_v1_STD_2026D98_noPU-v1/MINIAODSIM`
CMSSW version used: 14_1_X
Electron collection used : pat::Electron, slimmedElectrons  

<img width="840" alt="ele_barrel" src="https://github.com/cms-sw/cmssw/assets/5052706/3c0db269-737b-4ca6-bf89-4ebd195c374c">

**Comment by srimanob** [comment]
@cms-sw/muon-pog-l2 @cms-sw/egamma-pog-l2 
Do you have any update on this issues? Thanks.

**Comment by swagata87** [comment]
just to clarify, the issue is in muon only, not in egamma. 
so there is no action item for egamma.

**Comment by mmusich** [comment]
> Since the issue is with the electromagnetic component of the isolation, it would be nice if Egamma can check whether they see a similar trend in their isolation variable.

@rbhattacharya04 have you seen https://github.com/cms-sw/cmssw/issues/43858#issuecomment-1988267651 ? Do you have any reactions about the suggestion from @swagata87 ?

**Comment by JanFSchulte** [comment]
Hi everyone, 

on the muon side, we are working on a re-implementation of our ECAL-based isolation, ditching the no longer supported calo towers and switching to rec hits, following EGMs approach. It's a taking a bit longer than expected, but @24LopezR is working on it and will update here once a solution is implemented.

**Comment by 24LopezR** [comment]
Hi all,
So I computed the Ecal and Hcal isolation using RecHits instead of calotowers, and the weird structures seem to disappear.
[1] CaloTower Ecal iso, [2] RecHit Ecal iso
[3] CaloTower Hcal iso, [4] RecHit Hcal iso

[1] <img src="https://github.com/cms-sw/cmssw/assets/68101581/6e20483d-6ed4-477d-920a-b8b21dc72e44"  width="360"> [2] <img src="https://github.com/cms-sw/cmssw/assets/68101581/324a2b74-8860-4e19-92b8-aca749c08e84" width="360">
[3] <img src="https://github.com/cms-sw/cmssw/assets/68101581/e73346aa-9d62-44b7-bfba-cb3966a96c55" width="360"> [4] <img src="https://github.com/cms-sw/cmssw/assets/68101581/eb54f8a7-facc-4b7f-b4d7-c9d1031cd353" width="360">

Ceirtanly, the plots look more natural now. I am currently preparing the PR with this fix.

**Comment by mandrenguyen** [comment]
+1
This issue seems fixed by #44797

**Comment by srimanob** [comment]
+Upgrade

**Comment by mmusich** [comment]
> This issue seems fixed by https://github.com/cms-sw/cmssw/pull/44797

There is still the point at https://github.com/cms-sw/cmssw/pull/44797#issuecomment-2104224952, so I would prefer to keep this open until that's clarified.
